src/storage/src/table/state_table.rs

```diff
@@ -128,7 +128,9 @@ impl<S: StateStore> StateTable<S> {
 
     /// This function scans rows from the relational table.
     pub async fn iter(&self, epoch: u64) -> StorageResult<impl RowStream<'_>> {
-        let cell_based_bounds = (Unbounded, Unbounded);
+        let cell_based_start_key = Included(self.keyspace.key().to_vec());
+        let cell_based_end_key = Excluded(next_key(self.keyspace.key()));
+        let cell_based_bounds = (cell_based_start_key, cell_based_end_key);
```

Comment:
Please also fix iter_with_prefix. I have copied the code inside `self.iter()` to iter_with_prefix in #3126 .